src/Tests/Microsoft.NET.Restore.Tests/GivenThatWeWantToRestoreProjectsUsingNuGetConfigProperties.cs

```diff
@@ -156,5 +159,52 @@ private void ExtractNupkg(string nugetCache, string nupkg)
                 }
             }
         }
+
+        private class MockPackageDownloader : IPackageDownloader
```

Comment:
@nkolev92 @rohit21agrawal Can you provide guidance on how to implement an `IPackageDownloader` for a test like this?  I tried naively just implementing one method, and it looks like it's failing when it calls the `SignedPackageReader` getter.  What actually needs to be implemented here?

Reply:
we have a test implementation here: https://github.com/NuGet/NuGet.Client/blob/dev/test/NuGet.Core.Tests/NuGet.Packaging.Test/NugetPackageUtilTests.cs#L805

Hope that helps!